### DISCUSSION TITLE
blog/angular-router-standalone-apis

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#7] [General]
# blog/angular-router-standalone-apis

Let's take a look at Angulars brand new standalone Router APIs. Is it worth it? How much can we shake off the Routers bundle?

https://angularexperts.io/blog/angular-router-standalone-apis

**Comment by ashujainvi** [comment]
This article fills in the gap of the official doc where they don't walk you through how to convert existing `RouterModule` approach to standalone APIs. Great read, thanks for writing this!

It would be nice if we could update this article a step further and include the information about router unwrapping default imports for lazy loaded components as mentioned in the [ng15 release article](https://blog.angular.io/angular-v15-is-now-available-df7be7f2f4c8).

Before this change, to lazy load a standalone component you had to:

```
{
  path: 'lazy',
  loadComponent: () => import('./lazy-file').then(m => m.LazyComponent),
}
```

Now the router will look for a default export and if it finds it, use it automatically, which simplifies the route declaration to:

```
{
  path: 'lazy',
  loadComponent: () => import('./lazy-file'),
}
```

Note: You need to export component as `default` class for this to work.
